explainer.md

```diff
@@ -371,26 +337,110 @@ if(frameOfRef.bounds) {
+When `VRWebGLLayer.multiview` is true:
+- The UA may decide to back the framebuffer with a texture array, side-by-side texture or another implementation of the UA's choosing. This implementation decision must not have any impact how developers author their shaders or setup the WebGL context for rendering. 
+- When calling `VRPresentationFrame.getViewport()` with this type of `VRWebGLLayer`, the `VRView` parameter will be ignored.  All `VRView` objects use the same `VRWebGLViewport`.
 
 ```js
-vrDevice.requestSession().then(session => {
-  // Request dimensions needed for 1:1 output pixel ratio. 0.5 would request a
-  // half resolution buffer, values greater than 1.0 request a resolution that
-  // will be super-sampled. Passing 0.0 or leaving to optional scale factor off
-  // lets the UA decide what scale factor to use.
-  let sourceProperties = session.getSourceProperties(1.0);
+function setupWebGLLayer() {
+  return vrDevice.ensureContextCompatibility(gl).then(() => {
+    // VRWebGLLayer allows for the optional use of the WEBGL_multiview extension
+    vrSession.baseLayer = new VRWebGLLayer(vrSession, glCanvas, { multiview: true });
+  });
+}
 
-  glCanvas.width = sourceProperties.width;
-  glCanvas.height = sourceProperties.height;
+function onDrawFrame(vrFrame) {
+  // Do we have an active session?
+  if (vrSession) {
+    let pose = vrFrame.getDevicePose(vrFrameOfRef);
+    gl.bindFramebuffer(vrSession.baseLayer.framebuffer);
+    
+    if (vrSession.baseLayer.multiview) {
+      // When using the `WEBGL_multiview` extension, a `VRView` does not need
+      // to be supplied to `getViewport()` because all viewports are the same
+      let viewport = vrFrame.getViewport(vrSession.baseLayer);
+      gl.viewport(viewport.x, viewport.y, viewport.width, viewport.height);
+      drawMultiviewScene(vrFrame.views, pose);
+    } else {
+      for (let view in vrFrame.views) {
+        let viewport = vrFrame.getViewport(vrSession.baseLayer, view);
+        gl.viewport(viewport.x, viewport.y, viewport.width, viewport.height);
+        drawScene(view, pose);
+      }
+    }
 
-  // Do any other necessary setup
+    // Request the next VR callback
+    vrSession.requestVRPresentationFrame().then(onDrawFrame);
+    
+  } else {
+    // No session available, so render a default mono view.
+    gl.viewport(0, 0, glCanvas.width, glCanvas.height);
+    drawScene();
 
-  // Start render loop
-  vrSession.commit().then(OnFirstVRFrame);
-});
+    // Request the next window callback
+    window.requestAnimationFrame(onDrawFrame);
+  }
+}
+
+function drawMultiviewScene(sequence<VRView> views, VRPose pose) {
```

Comment:
As a lover of typed languages, I feel for you here. 😉 But... this IS Javascript, so lets drop the arg types.